source/blender/gpu/shaders/gpu_shader_material.glsl

```diff
@@ -1847,6 +1847,16 @@ void lamp_falloff_curve(float lampdist, sampler2D curvemap, float dist, out floa
 	visifac = texture2D(curvemap, vec2(dist / lampdist, 0.0)).x;
 }
 
+void lamp_falloff_invsquarecutoff(float radius, float dist, float cutoff, out float visifac)
+{
+	float d = max(dist - radius, 0.0);
+	float denom = d / radius + 1.0;
+ 	float att = 1.0 / (denom * denom);
+	att = (att - cutoff) / (1.0 - cutoff);
```

Comment:
`cutoff` is from 0 to 1 in the UI definition:
```c
RNA_def_property_range(prop, 0.0f, 1.0f);
```
And here it could cause 0 division. You could limit it to 0.99.

Reply:
I just to check other shaders functions and the indeterminations are handle inside the shader instead of clamping the property range.

Perhaps we can optimize the lamp_falloff_invsquarecutoff function as follow:
```c
void lamp_falloff_invsquarecutoff(float radius, float dist, float cutoff, out float visifac)  
{
	if (cutoff == 1.0) {
		visifac = 0.0;
	}
	else {
		float d = max(dist - radius, 0.0);  
		if (d == 0) {
			visifac = 1.0;
		}
		else {
			float denom = d / radius + 1.0;  
			float att = 1.0 / (denom * denom);  
			att = (att - cutoff) / (1.0 - cutoff);  
			att = max(att, 0.0);  
			visifac = att;
		}
	}
}
```
What do you think?

Reply:
Is there's an interest for the user to set a cutoff to 1 or a zero radius ?

I agree for the second check level, you can even remove the max a check d <= 0.0.

Also about the UI, you can define a new RNA attribute named radius that is the same as distance attribute.

Reply:
Ok, changes done